tools/vuln_valid/index.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+const Joi = require('joi');
+const Path = require('path');
+const Fs = require('fs');
+
+const vulnPath = Path.join(__dirname, '..', '..', 'vuln');
+const coreVulnPath = Path.join(vulnPath, 'core');
+const npmVulnPath = Path.join(vulnPath, 'npm');
+
+const coreModel = Joi.object().keys({
+    cve: Joi.array().items(Joi.string().regex(/CVE-\d{4}-\d+/)),
+    ref: Joi.string().uri().optional(),
+    vulnerable: Joi.string(),
```

Comment:
Would it be possible to create a joi extension that utilizes the `semver` module to validate these? If so, that would be cool.